video/out/vo_tct.c

```diff
@@ -101,37 +102,24 @@ static int rgb_to_x256(uint8_t r, uint8_t g, uint8_t b)
     return color_err <= gray_err ? 16 + color_index() : 232 + gray_index;
 }
 
-static void print_seq3(struct lut_item *lut, const char* prefix,
+static void print_seq3(bstr *frame, struct lut_item *lut, bstr prefix,
                        uint8_t r, uint8_t g, uint8_t b)
 {
-// The fwrite implementation is about 25% faster than the printf code
-// (even if we use *.s with the lut values), however,
-// on windows we need to use printf in order to translate escape sequences and
-// UTF8 output for the console.
-#ifndef _WIN32
-    fputs(prefix, stdout);
-    fwrite(lut[r].str, lut[r].width, 1, stdout);
-    fwrite(lut[g].str, lut[g].width, 1, stdout);
-    fwrite(lut[b].str, lut[b].width, 1, stdout);
-    fputc('m', stdout);
-#else
-    printf("%s;%d;%d;%dm", prefix, (int)r, (int)g, (int)b);
-#endif
+    bstr_xappend(NULL, frame, prefix);
```

Comment:
bstr_xappend calls a realloc and a memcpy. Do we want to do this five times, or would it make more sense to do a bstr_asprintf?

Reply:
There is already pregenerated lut for those values, specifically to avoid constant asprintf, which in itself involves some overhead.

realloc will happen only on first frame, the buffer is reused. memcpy will happen, but memory copy is the most basic operation we can do, asprintf would do a lot more.